mantis-server/mantis-server-agent/src/main/java/io/mantisrx/server/agent/TaskExecutor.java

```diff
@@ -384,6 +387,7 @@ public void startUp() throws Exception {
                 gateway
                     .registerTaskExecutor(taskExecutorRegistration)
                     .get(heartBeatTimeout.getSize(), heartBeatTimeout.getUnit());
+                registered = true;
```

Comment:
should you mark `registered = false` if connection doesn't go as planned? line:400

Reply:
line 400 is part of the initialization. The initial value of registered is set to false. So, that should handle the case when initialization fails